lib/internal/modules/esm/loader.js

```diff
@@ -71,6 +74,8 @@ class ESMLoader {
   #defaultLoad;
   #importMetaInitializer;
 
+  #dynamicImportCache = new SafeMap();
```

Comment:
Why would we create a new cache for dynamic import specifically? Shouldn’t these already be in the module map?

Shouldn’t static and dynamic import of the same input return the same result? Wouldn’t that mean they need to share the same cache?

Reply:
Indeed, dynamic and static imports must share the same cache.